githubauth/app.go

```diff
@@ -113,7 +130,14 @@ func NewApp[T *rsa.PrivateKey | string | []byte](appID string, privateKeyT T, op
 	}
 
 	for _, opt := range opts {
-		app = opt(app)
+		app, err = opt(app)
+		if err != nil {
+			return nil, fmt.Errorf("error when applying option: %w", err)
+		}
+	}
+
+	if app.signer == nil {
```

Comment:
It feels like we really want to have `NewApp` take a [`crypto.Signer`](https://pkg.go.dev/crypto#Signer) as a required argument. Then you could instantiate the signer outside of the call (by parsing an RSA key, setting up GCP KMS, etc).

Reply:
That's a reasonable suggestion, I'll try pivoting to that and see what it looks like.

Reply:
I think you are correct and this looks better.

Reply:
It happens sometimes